src/hooks/usePendingActions.ts

```diff
@@ -22,10 +22,19 @@ const usePendingActions = (chainId: string, safeAddress?: string): PendingAction
   const { safeAddress: currentSafeAddress } = useSafeInfo()
   const { page: currentSafeQueue } = useTxQueue()
   const isCurrentSafe = currentSafeAddress === safeAddress
+  const cachedQueue = useRef<TransactionListPage>()
 
-  const [loadedQueue] = useAsync<TransactionListPage>(() => {
+  const [loadedQueue] = useAsync<TransactionListPage | undefined>(async () => {
     if (isCurrentSafe || !safeAddress) return
-    return getTransactionQueue(chainId, safeAddress)
+
+    if (cachedQueue.current) {
+      return cachedQueue.current
+    }
+
+    const fetchedQueue = await getTransactionQueue(chainId, safeAddress)
+    cachedQueue.current = fetchedQueue
+
+    return fetchedQueue
```

Comment:
Looks good!

Would this work without async/await as well? So that if you scroll quickly, it wouldn't create simultaneous requests.

```suggestion
    if (!cachedQueue.current) {
      cachedQueue.current = getTransactionQueue(chainId, safeAddress) 
    }
    return cachedQueue.current
```

Reply:
Nice suggestion! We cache earlier and works like a charm